### DISCUSSION TITLE
How to inject `io.vertx.mutiny.core.http.HttpServerResponse`?

### DISCUSSION THREAD

**Discussion by fenix-hub** [quarkusio/quarkus#29844] [Q&A]
Hi,
I'm writing an http resource using Resteasy Reactive and I'd like to add a response header.
I'm following the [documentation about accessing contextual objects](https://quarkus.io/guides/resteasy-reactive#accessing-context-objects) and it declares that it is possible to inject `HttpServerResponse`, so I thought about using it to put an header.
However, it looks like it's not working, as Quarkus throws an exception
```log
Caused by: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.vertx.mutiny.core.http.HttpServerResponse and qualifiers
```

How do I set a response header still returning `Uni` and not a `RestResponse`?

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply]
I believe that returning a Uni<RestResponse> is the correct approach. @geoand?

**Reply by geoand** [reply]
Yes, using `Uni<RestResponse>` is the way to go

**Reply by fenix-hub** [reply]
Oh ok, so it is mandatory to return a RestResponse. But why the documentation mentions that it is possible to Inject HttpServerResponse and HttpServerRequest? Am I doing something wrong?

**Reply by geoand** [reply]
You can, but you should not
